### PR TITLE
[DIR-709] buttonbar needs to support label, update button component to have asLabel props

### PR DESCRIPTION
**Pull request by porleaaron**
### What's done

1. update the Button component to have props asLabel
2. update the ButtonBar component to handle the "label" children.
3. update the stories for Button and ButtonBar component
### What needs to be discussed

1. Does label have Hover effect as I have removed the click effect from button in case of asLabel
2. Any other UI requirements in case of label?

### CODE REVIEW

**Comment by stefan-kracht** [comment]
EDIT: Sorry, that comment was on the wrong PR :) 

Can you elaborate on the status of this PR? I see an implementation of multiple approaches that we discussed. So does our preferred one work? 

```typescript
<Button asChild>
  <label>Label</label>
</Button>
``` 

Also the `asLabel`, was meant to be `as="label"` and could be anything else as well (like `as="div"`). That surely was not clear, because you haven't seen it written down :) But this would only be our fallback solution if the first one would not work.

**Comment by stefan-kracht** [comment]
As soon as this is a draft PR it would be great if you give me some context of the status. It would be much easier for me to know what you would like me to check. E.g I see some TS errors and I don't know if they are intentional.

**Comment by stefan-kracht** [comment]
The code in the Button Component looks very good to me!! If that is what you would like me to check :)

**Comment by porleaaron** [comment]
This branch has now all we wanted at the beginning, you can have a review @stefan-kracht

File: src/design/ButtonBar/index.stories.tsx [hunk 1]
```diff
@@ -141,7 +175,7 @@ namespaced which would have a DX impact that is not worth it for now.
         </Button>
         <Tooltip>
           <TooltipTrigger asChild>
-            <div>
+            <div className="!border-0">
```
Comment (stefan-kracht) [inline src/design/ButtonBar/index.stories.tsx:178]: We somehow have to find a way to get rid of this with a tailwind css selector in the ButtonBar Component? Removing the border should all be handled by the ButtonBar. 


I think this also broke this part of our workflow editor

CleanShot 2023-08-03 at 09 18 40@2x

File: src/design/ButtonBar/index.tsx [hunk 2]
```diff
@@ -15,6 +15,15 @@ export const ButtonBar: React.FC<HTMLAttributes<HTMLDivElement>> = ({
       "[&_button]:active:outline-none",
       "[&_button]:border [&_button]:focus:ring-0 [&_button]:focus:ring-offset-0",
       "[&_button]:active:ring-0 [&_button]:active:ring-offset-0",
+
```
Comment (stefan-kracht) [inline src/design/ButtonBar/index.tsx:18]: Can you clean up the comments here? There are some empty l ines and a 
`//in case of label`  comment that I don't get.

File: src/design/ButtonBar/index.stories.tsx [hunk 3]
```diff
@@ -102,6 +102,35 @@ export const ButtonSizes = () => (
   </div>
 );
 
+export const AsChildButtons = () => (
+  <div className="flex flex-col gap-4">
+    <p>No Click, Hover effects on child components except Anchor</p>
+    <ButtonBar>
+      <Button asChild size="lg">
```
Comment (stefan-kracht) [inline src/design/ButtonBar/index.stories.tsx:109]: Could only one Label be enough here? I was just wondering what the difference of Label1 and Label2 is

File: src/design/ButtonBar/index.stories.tsx [hunk 4]
```diff
@@ -102,6 +102,35 @@ export const ButtonSizes = () => (
...
+        <GitBranchIcon /> Review
+      </Button>
+      <Button size="lg">
+        <SearchIcon /> Search
+      </Button>
+      <Button asChild size="lg">
+        <span>Span Tag</span>
+      </Button>
+      <Button variant="primary" asChild size="lg">
+        <span>No Hover</span>
```
Comment (stefan-kracht) [inline src/design/ButtonBar/index.stories.tsx:128]: What does "No Hover" mean here

**Review by stefan-kracht — CHANGES_REQUESTED** [review]
I added some comments and this part of the [Ticket](https://linear.app/direktiv/issue/DIR-709/buttonbar-needs-to-support-label) is still missing (the filters on the instances list need to be updated accordingly):
- [ ] change the buttons on every filter to be a label

File: src/design/ButtonBar/index.stories.tsx (hunk 4 above)
Comment (porleaaron) [reply src/design/ButtonBar/index.stories.tsx:128]: Primary button has the Hover Effect 
But when we apply asChild props, and the child is Span, then there is no hover effect
Only <a> child will have the hover effect, click effect
This is what I wanted to say in the story

File: src/design/ButtonBar/index.stories.tsx (hunk 3 above)
Comment (porleaaron) [reply src/design/ButtonBar/index.stories.tsx:109]: Yes, both labels are the same, we can remove one of them

File: src/design/ButtonBar/index.stories.tsx [hunk 5]
```diff
@@ -125,7 +148,8 @@ https://github.com/radix-ui/primitives/discussions/560
 TLDR; It could technically solved, but all state attributes would need to be 
 namespaced which would have a DX impact that is not worth it for now.
              */}
-            <div>
+
```
Comment (stefan-kracht) [inline src/design/ButtonBar/index.stories.tsx:151]: can you please remove empty `className` attributes?

**Comment by stefan-kracht** [comment]
Thanks @porleaaron, I quickly did this one task that was still missing

> I added some comments and this part of the [Ticket](https://linear.app/direktiv/issue/DIR-709/buttonbar-needs-to-support-label) is still missing (the filters on the instances list need to be updated accordingly):
> 
> * [ ]  change the buttons on every filter to be a label

**Review by stefan-kracht — APPROVED** [review]
